src/cli/README_DATASET.md

```diff
@@ -107,6 +107,108 @@ After the device successfully attaches to a Thread network, the device will retr
    Done
    ```
 
+### Using the Pending Operational Dataset for Delayed Dataset Updates
+
+The Pending Operational Dataset can be used for a delayed update of network parameters on all devices of a Thread Network.
+If certain Active Operational Dataset parameter(s) need to be changed, but the change would impact the connectivity of the network,
+delaying the update helps to let all devices receive the new parameter(s) before the update is applied. Examples of such
+parameters are the channel, PAN ID, certain Security Policy bits, or Network Key.
+
+The delay timer determines the time period after which the Pending Dataset takes effect and becomes the
+Active Dataset. The below example shows how a Pending Dataset with delay timer can be set at a Leader device.
+The Leader will initiate the distribution of the Pending Dataset to the rest of the devices in the network.
+
+Normally, an active Commissioner will set a new Pending Dataset. For testing purposes, we will do this in the example
+directly on the Leader using the CLI - so without using a Commissioner.
+
+1. Create a new Dataset in the dataset buffer, by copying the Active Dataset. Then change the channel number.
+
+   ```bash
+   > dataset init active
+   Done
+   > dataset activetimestamp 1696177379
+   Done
+   > dataset pendingtimestamp 1696177379
+   Done
+   > dataset channel 12
+   Done
+   ```
+
+2. Set the delay timer parameter to 5 minutes (300000 ms). Show the resulting Dataset that's ready to be used.
+
+   ```bash
+   > dataset delay 300000
+   Done
+   > dataset
+   Pending Timestamp: 1696177379
+   Active Timestamp: 1696177379
+   Channel: 12
+   Channel Mask: 0x07fff800
+   Delay: 300000
+   Ext PAN ID: 324a71d90cdc8345
+   Mesh Local Prefix: fd7d:da74:df5e:80c::/64
+   Network Key: be768535bac1b8d228960038311d6ca2
+   Network Name: OpenThread-bcaf
+   PAN ID: 0xbcaf
+   PSKc: e79b274ab22414a814ed5cce6a30be67
+   Security Policy: 672 onrc 0
+   Done
+   ```
+
+3. Commit the new Dataset as the Pending Dataset. This also starts the delay timer countdown. The Leader then starts the distribution of the Pending Dataset to other devices in the network.
+
+   ```bash
+   > dataset commit pending
+   Done
+   ```
+
+4. (Optional) Check that the delay timer is counting down, by displaying the Pending Dataset, after a few seconds.
```

Comment:
While it's optional, for the illustrative purposes of this example, we don't need to note it explicitly.  We can just say, "To verify that the delay timer is counting down, check the Pending Operational Dataset after a few seconds."